src/libraries/System.Private.CoreLib/src/System/Threading/WaitSubsystem.ThreadWaitInfo.Unix.cs

```diff
@@ -395,7 +395,7 @@ public static void Sleep(int timeoutMilliseconds, bool interruptible)
 
                 if (timeoutMilliseconds == 0)
                 {
-                    if (interruptible && Thread.CurrentThread.WaitInfo.CheckAndResetPendingInterrupt_NotLocked)
+                    if (interruptible && Thread.CurrentThread._waitInfo.CheckAndResetPendingInterrupt_NotLocked)
```

Comment:
Having an internal WaitInfo property is fine imho.

Reply:
I wasn't sure which convention was preferred in dotnet/runtime; it's easy to change in either direction.

Reply:
@marek-safar do you know which convention is preferred?

Reply:
Either is ok.  Properties are a tad more expensive for the JIT as they need to be inlined to get the right perf.  But especially between components it's often worth it for the flexibility of being able to easily change the property's implementation without having to also find and update all call sites.